pkg/helmshim/helm_test.go

```diff
@@ -16,12 +17,38 @@ func TestRenderChart(t *testing.T) {
 	input := bytes.NewBufferString(`{ "items": [{ "apiVersion": "v1", "kind": "ConfigMap", "metadata": { "name": "test-cm", "annotations": { "eno.azure.io/input-key": "foo" } } }] }`)
 	i, err := function.NewInputReader(input)
 	require.NoError(t, err)
-
 	err = RenderChart(WithInputReader(i), WithOutputWriter(o), WithChartPath("fixtures/basic-chart"))
 	require.NoError(t, err)
+	err = o.Write()
+	require.NoError(t, err)
 	assert.Equal(t, "{\"apiVersion\":\"config.kubernetes.io/v1\",\"kind\":\"ResourceList\",\"items\":[{\"apiVersion\":\"v1\",\"data\":{\"input\":\"{\\\"apiVersion\\\":\\\"v1\\\",\\\"kind\\\":\\\"ConfigMap\\\",\\\"metadata\\\":{\\\"annotations\\\":{\\\"eno.azure.io/input-key\\\":\\\"foo\\\"},\\\"name\\\":\\\"test-cm\\\"}}\",\"inputResourceName\":\"test-cm\",\"some\":\"value\"},\"kind\":\"ConfigMap\",\"metadata\":{\"name\":null}},{\"apiVersion\":\"somegroup.io/v9001\",\"kind\":\"ATypeNotKnownByTheScheme\",\"metadata\":{\"name\":\"foo\"}}]}\n", output.String())
 }
 
+func TestRenderChartWithDefaultOutputWriter(t *testing.T) {
+	// Save the original stdout and redirect it to a pipe.
+	oldStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	input := bytes.NewBufferString(`{ "items": [{ "apiVersion": "v1", "kind": "ConfigMap", "metadata": { "name": "test-cm", "annotations": { "eno.azure.io/input-key": "foo" } } }] }`)
+	i, err := function.NewInputReader(input)
+	require.NoError(t, err)
+	// Do not provide an output writer and use the default output writer.
+	err = RenderChart(WithInputReader(i), WithChartPath("fixtures/basic-chart"))
+	require.NoError(t, err)
+
+	// Close the writer and capture the output from reader.
+	w.Close()
+	var buf bytes.Buffer
+	buf.ReadFrom(r)
+	output := buf.String()
+	// Restore the original stdout.
+	os.Stdout = oldStdout
```

Comment:
We should defer this right after overriding it so the value will be reset even if the test fails

Reply:
update! good call that the test fails will cause the stdout didn't get restore back